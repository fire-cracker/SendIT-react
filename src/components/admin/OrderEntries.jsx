import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';


const OrderEntries = ({ showPending, orders }) => {
  const order = orders.map((item, index) => {
    const {
      fromAddress, toAddress, weight, price, orderStatus, createdAt, presentLocation
    } = item;

    const date = format(
      new Date(createdAt),
      'MM/DD/YYYY, hh:mm a'
    );
    return (
      <tr key={parseInt(index.toString(), 7)}>
        <td>{fromAddress}</td>
        <td>{toAddress}</td>
        <td>{weight}</td>
        <td>
          &#8358;
          {price}
        </td>
        <td>{date}</td>
        <td>{orderStatus}</td>
        {showPending ? null : (<td>{presentLocation}</td>)}
      </tr>
    );
  });

  return (
    order
  );
};

OrderEntries.propTypes = {
  orders: PropTypes.array,
};

export default OrderEntries;
