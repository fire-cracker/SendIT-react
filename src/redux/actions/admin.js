import axios from '../../utils/axiosConfig';
import { GET_ALL_ORDERS_REQUEST, GET_ALL_ORDERS_SUCCESS } from './actionTypes';

const api = process.env.API_ROOT_URL;

export const getAllOrdersRequest = () => ({
  type: GET_ALL_ORDERS_REQUEST
});

export const getAllOrdersSuccess = data => ({
  type: GET_ALL_ORDERS_SUCCESS,
  allOrders: data
});

export const getAllOrders = () => async (dispatch) => {
  try {
    dispatch(getAllOrdersRequest());
    const { data } = await axios.get(`${api}/parcels`);
    dispatch(getAllOrdersSuccess(data));
  } catch (error) {
    throw error;
  }
};
