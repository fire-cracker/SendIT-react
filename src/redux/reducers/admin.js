import {
  GET_ALL_ORDERS_REQUEST,
  GET_ALL_ORDERS_SUCCESS,
} from '../actions/actionTypes';


const orderInitialState = {
  allOrders: []
};
export default (state = orderInitialState, action) => {
  switch (action.type) {
    case GET_ALL_ORDERS_REQUEST:
      return {
        ...state
      };

    case GET_ALL_ORDERS_SUCCESS: {
      const { orders: { orders } } = action;
      return {
        ...state,
        allOrders: orders,
      };
    }
    default:
      return state;
  }
};
