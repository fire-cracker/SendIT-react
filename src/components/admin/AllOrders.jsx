import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';

import { getAllOrders } from '../../redux/actions/admin';
import OrderEntries from './OrderEntries';
import Header from '../header';
import './style.css';

export class GetAllOrders extends Component {
  state = {
    showNew: false,
    showPending: true,
    showDelivered: true,
    status: 'New'
  }

  componentDidMount() {
    const { getAllOrders, history } = this.props;
    getAllOrders()
      .then()
      .catch((error) => {
        const { response, response: { status } } = error;
        if (response && status === 401) {
          history.push('/');
          return toast.error('Your session has expired. You need to login');
        }
        toast.error('Unknown error');
      });
  }

  componentDidUpdate() {
    const { history, login: { isLoggedIn } } = this.props;
    if (isLoggedIn === false) {
      history.push('/');
    }
  }

  toggleDiv = (status, showingDiv) => {
    this.setState(() => ({
      showNew: true,
      showPending: true,
      showDelivered: true,
      [showingDiv]: false,
      status
    }));
  };

  render() {
    const { allOrders: { allOrders } } = this.props;
    const {
      showNew, showPending, showDelivered, status
    } = this.state;

    const orderEntries = allOrders.filter(order => order.orderStatus === status);

    return (
      <div>
        <div>
          <Header />
        </div>
        <div id="informationFields">
          <div id="preview" className="preview">
            <h1> Hi Admin! Track Your Orders </h1>

            <div className="navTabs" id="navTabs">
              <a
                className={showNew ? 'navLink' : 'navLink active'}
                onClick={() => this.toggleDiv('New', 'showNew')}
              >
                NEW ORDERS
              </a>
              <a
                className={showPending ? 'navLink' : 'navLink active'}
                onClick={() => this.toggleDiv('Pending', 'showPending')}
              >
                PENDING ORDERS
              </a>
              <a
                className={showDelivered ? 'navLink' : 'navLink active'}
                onClick={() => this.toggleDiv('Delivered', 'showDelivered')}
              >
                DELIVERED ORDERS&nbsp;
              </a>
            </div>

            <div id="information">

              <div
                id="shipmentOrder"
                className="newPending-active"
              >
                { orderEntries.length
                  ? (
                    <div className="order-table">
                      <table>
                        <thead>
                          <tr>
                            <th>Pickup Location</th>
                            <th>Destination</th>
                            <th>weight(kg)</th>
                            <th>Price</th>
                            <th>Date of Order</th>
                            <th>Status</th>
                            {showPending ? null : (<th>PresentLocation</th>)}
                          </tr>

                          <OrderEntries showPending={showPending} orders={orderEntries} />

                        </thead>
                      </table>
                    </div>
                  )
                  : (
                    <p className="no__order">No Order</p>
                  )
                  }
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

GetAllOrders.propTypes = {
  getAllOrders: PropTypes.func,
  allOrders: PropTypes.object,
  history: PropTypes.object,
  login: PropTypes.object
};
const mapStateToProps = ({ login, allOrders }) => ({ login, allOrders });

const mapDispatchToProps = {
  getAllOrders
};

export default connect(mapStateToProps, mapDispatchToProps)(GetAllOrders);
