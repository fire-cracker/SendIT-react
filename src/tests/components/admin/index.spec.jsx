import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import { GetAllOrders } from '../../../components/admin/AllOrders';
import { allOrders } from '../../mock/orders';

jest.mock('../../../utils/axiosConfig');
jest.mock('react-toastify');
jest.mock('../../../redux/actions/createOrder');

describe('Tests for the track Page', () => {
  const props = {
    getAllOrders: jest.fn().mockResolvedValue({}),
    allOrders: { allOrders },
    login: {
      isLoggedIn: true
    }
  };
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<MemoryRouter><GetAllOrders {...props} /></MemoryRouter>);
  });

  it('Should render the track page', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.find('.navLink').at(0).simulate('click');
    wrapper.find('.navLink').at(1).simulate('click');
    wrapper.find('.navLink').at(2).simulate('click');
  });

  it('should throw error 401 if request is incorrect', () => {
    const props = {
      getAllOrders: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 401
        }
      })),
      allOrders: { allOrders }
    };
    wrapper = mount(<MemoryRouter><GetAllOrders {...props} /></MemoryRouter>);
  });

  it('should throw error if request is not sucessful', () => {
    const props = {
      getAllOrders: jest.fn().mockImplementation(() => Promise.reject({
        response: {
          status: 500
        }
      })),
      allOrders: { allOrders }
    };
    wrapper = mount(<MemoryRouter><GetAllOrders {...props} /></MemoryRouter>);
  });
});
