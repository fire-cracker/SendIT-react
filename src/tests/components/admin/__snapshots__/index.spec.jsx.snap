// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the track Page Should render the track page 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "ur37ke",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "ur37ke",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <GetAllOrders
      allOrders={
        Object {
          "allOrders": Array [
            Object {
              "fromAddress": "Oyedeji Haven, Jagun Old Ife Road Ibadan",
              "toAddress": "Oyedeji Haven, Jagun Old Ife Road Ibadan",
              "weight": "2",
            },
            Object {
              "fromAddress": "Oyedeji Haven, Jagun Old Ife Road Ibadan",
              "toAddress": "Oyedeji Haven, Jagun Old Ife Road Ibadan",
              "weight": "2",
            },
          ],
        }
      }
      getAllOrders={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Promise {},
            },
          ],
        }
      }
      login={
        Object {
          "isLoggedIn": true,
        }
      }
    >
      <div>
        <div>
          <Header>
            <div
              className="header"
              id="header"
            >
              <ToastContainer
                autoClose={5000}
                bodyClassName={null}
                className={null}
                closeButton={
                  <CloseButton
                    ariaLabel="close"
                  />
                }
                closeOnClick={true}
                draggable={true}
                draggablePercent={80}
                hideProgressBar={false}
                newestOnTop={false}
                pauseOnFocusLoss={true}
                pauseOnHover={true}
                position="top-right"
                progressClassName={null}
                progressStyle={null}
                rtl={false}
                style={null}
                toastClassName={null}
                transition={[MockFunction]}
              />
              <header
                className="navBar"
                id="navBar"
              >
                <div
                  id="headerContent"
                >
                  <div
                    className="navigation"
                    id="navigation"
                  >
                    <Link
                      id="logout"
                      replace={false}
                      to="/"
                    >
                      <a
                        href="/"
                        id="logout"
                        onClick={[Function]}
                      >
                        Logout
                      </a>
                    </Link>
                    <Link
                      id="profile"
                      replace={false}
                      to="/profile"
                    >
                      <a
                        href="/profile"
                        id="profile"
                        onClick={[Function]}
                      >
                        Profile
                      </a>
                    </Link>
                    <Link
                      id="active"
                      replace={false}
                      to="/createOrder"
                    >
                      <a
                        href="/createOrder"
                        id="active"
                        onClick={[Function]}
                      >
                        Create Order
                      </a>
                    </Link>
                    <Link
                      id="track"
                      replace={false}
                      to="/track"
                    >
                      <a
                        href="/track"
                        id="track"
                        onClick={[Function]}
                      >
                        Track
                      </a>
                    </Link>
                    <a
                      className="icon"
                      id="icon"
                      onClick={[Function]}
                    >
                      <i
                        className="fa fa-bars"
                      />
                    </a>
                  </div>
                </div>
                <div
                  className="blink_caption"
                  href="home.html"
                >
                  <div
                    className="blink"
                    id="blin"
                  >
                    Just SendIT!!!
                  </div>
                </div>
              </header>
            </div>
          </Header>
        </div>
        <div
          id="informationFields"
        >
          <div
            className="preview"
            id="preview"
          >
            <h1>
               Hi Admin! Track Your Orders 
            </h1>
            <div
              className="navTabs"
              id="navTabs"
            >
              <a
                className="navLink active"
                onClick={[Function]}
              >
                NEW ORDERS
              </a>
              <a
                className="navLink"
                onClick={[Function]}
              >
                PENDING ORDERS
              </a>
              <a
                className="navLink"
                onClick={[Function]}
              >
                DELIVERED ORDERSÂ 
              </a>
            </div>
            <div
              id="information"
            >
              <div
                className="newPending-active"
                id="shipmentOrder"
              >
                <table>
                  <thead>
                    <tr>
                      <th>
                        Pickup Location
                      </th>
                      <th>
                        Destination
                      </th>
                      <th>
                        weight(kg)
                      </th>
                      <th>
                        Price
                      </th>
                      <th>
                        Date of Order
                      </th>
                      <th>
                        Status
                      </th>
                    </tr>
                    <OrderEntries
                      orders={Array []}
                      showPending={true}
                    />
                  </thead>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </GetAllOrders>
  </Router>
</MemoryRouter>
`;
